### ISSUE TITLE
Retuning .should.be.resolved logs UnhandledPromiseRejectionWarning

### ISSUE THREAD

**Issue by Arial7** [domenic/chai-as-promised#168]
Hey guys. I have the following Test:

```javascript
describe("Tests", function() {
        it("Should be resolved", () => { return Promise.resolve().should.be.resolved });
        it("Should be rejected", () => { return Promise.reject().should.be.rejected });
        it("Should be resolved too", () => { return (Promise.reject()).should.be.resolved });
    });
```

And the following output:

```
Tests
      ✓ Should be resolved
      ✓ Should be rejected
      ✓ Should be resolved too
(node:8527) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 2): undefined
```

I would expect test 3 to fail though!
I am using the default nodejs es6 Promises. With those, it should definitively work.

Thanks for help!

**Comment by meeber** [comment]
@Arial7 The syntax is `.should.be.fulfilled` instead of `.should.be.resolved`.

Unfortunately, the current version of Chai has no way to detect that an invalid assertion is being made, so it just passes silently. This is a horrible side-effect of property-based assertions which is being addressed in the upcoming version of Chai (4.0) with ES6 proxies to alert the user of an invalid assertion.

**Comment by Arial7** [comment]
@meeber 
I am so sorry 😅 I should definitively read the docs more carefully!
Cannot wait for Chai 4.0 then 😄
